Bottom: e640b8f7fa66da3d920386f77481b818a3544984
Top:    a6e990f8eb92cc0a82a0e5574b6922189717d9c8
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-31 16:31:30 -0500

Generate new ssh key on first run


---
diff --git a/docker/buildbot-master/buildbot/start.sh b/docker/buildbot-master/buildbot/start.sh
index 71002f8b..85c28248 100755
--- a/docker/buildbot-master/buildbot/start.sh
+++ b/docker/buildbot-master/buildbot/start.sh
@@ -1,10 +1,12 @@
 #!/bin/sh
 
+SSH_BITS=${SSH_BITS-"2048"}
+SSH_TYPE=${SSH_TYPE-"rsa"}
+SSH_FILE=${SSH_FILE-"$HOME/.ssh/id_rsa"}
+
 BUILDBOT_DATA=${BUILDBOT_DATA-"/buildbot"}
 
-mkdir -p ${HOME}/.ssh
-cp -R /run/secrets/ssh/* ${HOME}/.ssh
-chmod -R 600 ${HOME}/.ssh/*
+ssh-keygen -q -b ${SSH_BITS} -t ${SSH_TYPE} -N '' -f ${SSH_FILE} 0>&- 1>/dev/null 2>/dev/null
 
 ln -v -s -f ${HOME}/buildbot.tac ${HOME}/master.cfg ${HOME}/*.py $BUILDBOT_DATA
